src/Bridge/Doctrine/Orm/Extension/FilterEagerLoadingExtension.php

```diff
@@ -79,9 +81,14 @@ public function applyToCollection(QueryBuilder $queryBuilder, QueryNameGenerator
                 $in = $this->getQueryBuilderWithNewAliases($queryBuilder, $queryNameGenerator, $originAlias, $replacementAlias);
                 $in->select("IDENTITY($replacementAlias.$identifier)");
                 $queryBuilderClone->andWhere($queryBuilderClone->expr()->in("$originAlias.$identifier", $in->getDQL()));
+                $changedWhereClause = true;
             }
         }
 
+        if (false === $changedWhereClause) {
```

Comment:
Can't you just check if the `where` clause of the query builder is empty instead of introducing a new variable?

Reply:
There's already a check for that [here](https://github.com/soyuka/core/blob/776f62d99219c4aa5c2ffa517310d549ee83734e/src/Bridge/Doctrine/Orm/Extension/FilterEagerLoadingExtension.php#L52). This is really an edge case.